media/server/gstplayer/source/GstGenericPlayer.cpp

```diff
@@ -615,13 +620,18 @@ void GstGenericPlayer::updateVideoCaps(int32_t width, int32_t height, Fraction f
             m_gstWrapper->gstCapsSetSimple(newCaps, "framerate", GST_TYPE_FRACTION, frameRate.numerator,
                                            frameRate.denominator, NULL);
         }
-        if (codecData)
+        if (codecData && CodecDataType::BUFFER == codecData->type)
```

Comment:
nit pick: I think it would be worth to extract codecData changing to a function (e.g. _bool GstGenericPlayer::setCodecData_(const std::shared_ptr<CodecData> &codecData)) since most of the code is the same (except capsChanged part in audio) and it's complicated enough to be worth to extract it.

Reply:
I will change it in next commit.